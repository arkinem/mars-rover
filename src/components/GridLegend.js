import React from "react";
import styled from "styled-components";
import { colors } from "../helpers/style";

export default () => {
  return (
    <Container>
      <LegendUnit>
        <ColorBox color={colors.grid.startPosition} />
        Start Position
      </LegendUnit>
      <LegendUnit>
        <ColorBox color={colors.grid.finishPosition} />
        End Position
      </LegendUnit>
      <LegendUnit>
        <ColorBox color={colors.grid.path} />
        Selected Rover Path
      </LegendUnit>
      <LegendUnit>
        <ColorBox color={colors.grid.otherRovers} />
        Other Rovers
      </LegendUnit>
    </Container>
  );
};

const Container = styled.div`
  position: relative;
  width: 100%;
  padding: 12px;
  display: flex;
  flex-wrap: wrap;
`;

const LegendUnit = styled.div`
  display: flex;
  flex-basis: 50%;
  margin-bottom: 8px;
  font-size: 14px;
`;

const ColorBox = styled.div`
  margin-right: 12px;
  height: 14px;
  width: 14px;
  border: 1px solid ${colors.border};
  background-color: ${({ color }) => color};
`;
