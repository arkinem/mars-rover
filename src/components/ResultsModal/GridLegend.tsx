import React, { FunctionComponent } from "react";
import styled from "styled-components";
import { LegendUnit } from "./LegendUnit";
import { colors } from "../../constants/style";
import strings from "../../constants/strings";

export const GridLegend: FunctionComponent<{}> = () => (
  <Container>
    <LegendUnit
      label={strings.legend.startPosition}
      color={colors.grid.startPosition}
    />
    <LegendUnit
      label={strings.legend.endPosition}
      color={colors.grid.finishPosition}
    />
    <LegendUnit
      label={strings.legend.selectedRoverPath}
      color={colors.grid.path}
    />
    <LegendUnit
      label={strings.legend.otherRovers}
      color={colors.grid.otherRovers}
    />
  </Container>
);

const Container = styled.div`
  position: relative;
  width: 100%;
  padding: 12px;
  display: flex;
  flex-wrap: wrap;
`;
