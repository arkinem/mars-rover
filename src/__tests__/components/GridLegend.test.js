import React from "react";
import renderer from "react-test-renderer";
import GridLegend from "../../components/ResultsModal/GridLegend";

describe("GridLegend", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<GridLegend />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
