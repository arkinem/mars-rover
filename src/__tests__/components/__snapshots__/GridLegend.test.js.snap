// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridLegend renders correctly 1`] = `
<div
  className="sc-AxjAm bdwZFi"
>
  <div
    className="sc-AxirZ icCRxH"
  >
    <div
      className="sc-AxiKw bToHce"
      color="#2b580c"
    />
    Start Position
  </div>
  <div
    className="sc-AxirZ icCRxH"
  >
    <div
      className="sc-AxiKw isfYCT"
      color="#900c3f"
    />
    End Position
  </div>
  <div
    className="sc-AxirZ icCRxH"
  >
    <div
      className="sc-AxiKw tWCEw"
      color="#639a67"
    />
    Selected Rover Path
  </div>
  <div
    className="sc-AxirZ icCRxH"
  >
    <div
      className="sc-AxiKw ghCWqq"
      color="#ff5733"
    />
    Other Rovers
  </div>
</div>
`;
